src/storage/src/hummock/state_store.rs

```diff
@@ -300,11 +301,17 @@ impl HummockStorage {
             if level.table_infos.is_empty() {
                 continue;
             }
-            let table_infos = prune_ssts(level.table_infos.iter(), &(key..=key));
-            for table_info in table_infos {
+            let table_info_idx = level
+                .table_infos
+                .partition_point(|table| {
+                    let ord = user_key(&table.key_range.as_ref().unwrap().left).cmp(key.as_ref());
+                    ord == Ordering::Less || ord == Ordering::Equal
+                })
+                .saturating_sub(1);
+            if table_info_idx < level.table_infos.len() {
```

Comment:
If the index returned from `partition_point` is 0, we can skip this level and avoid further filtering. The index to do further filtering will be non-zero, and therefore we don't need to do `saturating_sub`. 

The maximum index returned from `partition_point` is `table_infos.len()`, and after minus 1, the index must be less than `table_infos.len()`, and this comparison is unnecessary. 

After we get the table_index, we can further check whether the key falls in min-max of table info. If it doesn't, we can skip this level.

Reply:
> If the index returned from `partition_point` is 0, we can skip this level and avoid further filtering. The index to do further filtering will be non-zero, and therefore we don't need to do `saturating_sub`.
> 
> The maximum index returned from `partition_point` is `table_infos.len()`, and after minus 1, the index must be less than `table_infos.len()`, and this comparison is unnecessary.
> 
> After we get the table_index, we can further check whether the key falls in min-max of table info. If it doesn't, we can skip this level.

ok , fixed it